Demo/Demo/Networking/DemoMerchantAPI.swift

```diff
@@ -67,4 +60,50 @@ final class DemoMerchantAPI {
     private func buildBaseURL(with endpoint: String) -> URL? {
         URL(string: DemoSettings.environment.baseURL + endpoint)
     }
+
+    public func getAccessToken(environment: Demo.Environment) async -> String? {
+        guard let token = self.accessToken else {
+            self.accessToken = await fetchAccessToken(environment: environment)
+            return self.accessToken
+        }
+        return token
+    }
+
+    private func fetchAccessToken(environment: Demo.Environment) async -> String? {
+        do {
+            let accessTokenRequest = AccessTokenRequest()
+            guard let request = createUrlRequest(
+                accessTokenRequest: accessTokenRequest, environment: environment) else {
+                throw URLResponseError.dataParsingError
+            }
+            let (data, response) = try await URLSession.shared.performRequest(with: request)
+            guard let response = response as? HTTPURLResponse else {
+                throw URLResponseError.networkConnectionError
+            }
+            switch response.statusCode {
+            case 200..<300:
+                let accessTokenResponse: AccessTokenResponse = try parse(from: data)
+                return accessTokenResponse.accessToken
+            default: throw URLResponseError.serverError
+            }
+        } catch {
+            print("Error in fetching token")
+            return nil
+        }
+    }
+
+    private func createUrlRequest(accessTokenRequest: AccessTokenRequest, environment: Demo.Environment) -> URLRequest? {
+        var completeUrl = environment.baseURL
+        completeUrl.append(contentsOf: accessTokenRequest.path)
+        guard let url = URL(string: completeUrl) else {
+            return nil
```

Comment:
Seems like we handle the error portion for this failing on line 77 - since these are private helper functions, I feel like it's probably fine either way? This function can only fail for one reason, which is the `URL` initialization failure, so I think returning `nil` in this case is kind of okay, as we aren't going to be confused between `nil` representing multiple error cases. 

We could change the error on line 77 to be a little clearer though? Right now it is: `URLResponseError.dataParsingError` - could change it to something like `URLResponseError.urlInitializationError` or something that indicates its related to the URL string having a problem?

Reply:
changed error handling part, check again